eng/mgmt/automation/parameters.py

```diff
@@ -15,8 +15,8 @@
 MAVEN_URL = 'https://repo1.maven.org/maven2/{group_id}/{artifact_id}/{version}/{artifact_id}-{version}.jar'
 
 SDK_ROOT = '../../../'  # related to file dir
-AUTOREST_CORE_VERSION = '3.8.1'
-AUTOREST_JAVA = '@autorest/java@4.0.59'
+AUTOREST_CORE_VERSION = '3.8.4'
```

Comment:
@weidongxu-microsoft have we considered adding some automation to update these versions when we update the autorest version? We do that for .NET already and might help with some of these updates.

Reply:
Hi @weshaggard ,

We certainly can (I did think about it, save these PRs, and maybe just 10 more lines in script).

On rare case, the latest autorest.core be buggy (seen once or twice, though patch version is relatively safe), and latest autorest.java too (hope I can say never happen, but...).
Therefore, it is still safer to do this manually, it is only once or twice per month. We will do the switch when we are more confident (or if you feel it better to be consistent across repo, we can switch now, as mentioned, bug on latest version is still rare).

Reply:
I would expect the automation to still create a PR which would cover the testing aspect. So it would still require a review and merge but it would just limit one extra manual step for you guys.

Reply:
Thanks Wes, it is a great idea, we can have autorest.java release automatically add PR on SDK repo.

The testing part would likely be different from .NET. I think .NET will regen all the generated code in SDK repo with the new codegen and run SDK tests on it.
But Java does not have this, as many dev actually modify the generated code after codegen (which makes the regen impossible), and hence no such regen step build-in in Java SDK repo.

Instead, we had tests on 200 control-plane SDK daily (release or not) at https://github.com/Azure/autorest.java/wiki/Fluent-Codegen-Report, besides all the CIs in codegen (UT or integrated).
And smaller number of DPGs as well.
But these tests are weaker than .NET as it does not include recorded tests in major data-plane SDKs (as they are not DPG and not control-plane).

On what we have control, we might let the new codegen regen the (about 10) Ring0 control-plane SDK, and use existing recorded tests as additional check.

Reply:
Just in case here is a pointer to what .NET is doing https://github.com/Azure/autorest.csharp/blob/feature/v3/eng/pipelines/build.yml#L126. They run tests in their autorest pipeline and then have it submit a PR to the azure-sdk-for-net repo with the version increment after it is published.

Reply:
log task https://github.com/Azure/autorest.java/issues/1504